downstairs/src/lib.rs

```diff
@@ -1300,6 +1335,21 @@ impl Downstairs {
         ds_id: u64,
         work: IOop,
     ) -> Result<()> {
+        // The Upstairs will send Flushes periodically, even in read only mode
+        // we have to accept them. But read-only should never accept writes!
+        if self.read_only
+            && matches!(
+                work,
+                IOop::Write {
```

Comment:
`IOop::WriteUnwritten` should be treated like a `IOop::Write`

Reply:
ah, missed that - I made this a match on the type instead so that in the future if someone adds an `IOop` they will have to add a variant here. Fixed in 56f92bf